src/MoBi.Presentation/Tasks/AbstractMergeManager.cs

```diff
@@ -1,86 +1,40 @@
    {
-      protected Action<T> _removeAction;
-      protected Action<T> _addAction;
-      protected Action _cancelAction;
-      private readonly IMapper<T, ObjectBaseSummaryDTO> _dtoMapper;
-      protected Action<T, string> _cloneAction;
-      protected Action<T, T> _mergeAction;
-      private readonly IApplicationController _applicationController;
-      private readonly INameCorrector _nameCorrector;
-      private readonly IMoBiContext _context;
-      protected bool _mergeOptionEnabled = false;
-      protected bool _cloneOptionEnabled = false;
+      /// <summary>
+      ///    Manages the conflict of the two caches
+      /// </summary>
+      /// <param name="merge">The cache being merged into the target</param>
+      /// <param name="target">The target cache of the merge</param>
+      /// <param name="mergeConflictOption">The default option when merging items that are equivalent</param>
+      /// <param name="areElementsEquivalent">
+      ///    Optionally specify a Func that will detect for equivalence between two conflicting
+      ///    elements
+      /// </param>
+      void Merge(ICache<string, T> merge, ICache<string, T> target, MergeConflictOptions mergeConflictOption, Func<T, T, bool> areElementsEquivalent = null);
+   }
 
-      protected AbstractMergeManager(IApplicationController applicationController, INameCorrector nameCorrector, IMapper<T, ObjectBaseSummaryDTO> dtoMapper, IMoBiContext context)
-      {
-         _applicationController = applicationController;
-         _nameCorrector = nameCorrector;
-         _addAction = element => { };
-         _removeAction = element => { };
-         _cancelAction = () => { };
-         _cloneAction = (element, originalName) => { };
-         _mergeAction = (target, merge) => { };
-         _dtoMapper = dtoMapper;
-         _context = context;
-      }
+   public abstract class AbstractMergeManager<T> : IMergeManager<T> where T : class, IObjectBase
+   {
+      protected Action<T> _removeAction = element => { };
+      protected Action<T> _addAction = element => { };
 
-      public virtual void Merge(ICache<string, T> merge, ICache<string, T> target, Func<T, T, bool> areElementsEquivalent = null, MergeConflictOptions defaultOption = MergeConflictOptions.SkipOnce)
+      public virtual void Merge(ICache<string, T> merge, ICache<string, T> target, MergeConflictOptions mergeConflictOption, Func<T, T, bool> areElementsEquivalent = null)
       {
-         var resolved = 0;
          var conflictResolution = areElementsEquivalent ?? ((x, y) => false);
          var conflictingElements = GetConflictingElements(merge, target, conflictResolution).ToList();
 
-         var option = defaultOption;
          foreach (var key in conflictingElements)
          {
-            resolved++;
-            if (!option.IsAppliedToAll())
-               option = prompt(merge[key], target[key], conflictingElements.Count() - resolved);
```

Comment:
This could never have happened since we only ever call with `SkipAll` or `ReplaceAll`. There would never be a prompt, and since there's no prompt, we don't need most of the stuff to manage merge conflicts.